assignments/array-methods.js

```diff
@@ -54,30 +54,61 @@ const runners = [{"id":1,"first_name":"Charmain","last_name":"Seiler","email":"c
+// the dot notation version is element.key, so I want element.first_name and element.last_name;
+// Also I want to use backtick notation for the names.
+// So `${first_name} ${last_name}`
+// let fullName = runners.map(names => {return `${names.first_name} ${names.last_name}`});
+// Logging with a join to not have my node scroll horribly.
+// console.log('Challenge 1 Answer: ' + fullName.join(','));
+
+// OH WHOOPS SHOULD HAVE USED FOR EACH k, k, I can fix this.
 let fullName = [];
-console.log(fullName);
+runners.forEach((names) => {fullName.push(`${names.first_name} ${names.last_name}`)});
+// Logging with a join to not have my node scroll horribly.
+console.log('Challenge 1 Answer: ' + fullName.join(','));
 
 // ==== Challenge 2: Use .map() ====
 // The event director needs to have all the runner's first names converted to uppercase because the director BECAME DRUNK WITH POWER. Convert each first name into all caps and log the result
-let allCaps = [];
-console.log(allCaps); 
+
+// Similar thought process engage: .map, pull each first_name, .toUpperCase it.
+// backticks are more optional here but I'm using them so I get comfortable.
+let allCaps = runners.map((NAME) => {return `${NAME.first_name.toUpperCase()}`});
+// Logging with a join to not have my node scroll horribly.
+console.log('Challenge 2 Answer: ' + allCaps.join(','));
 
 // ==== Challenge 3: Use .filter() ====
 // The large shirts won't be available for the event due to an ordering issue.  Get a list of runners with large sized shirts so they can choose a different size. Return an array named largeShirts that contains information about the runners that have a shirt size of L and log the result
-let largeShirts = [];
-console.log(largeShirts);
+let largeShirts = runners.filter((backorder) => {
+  return backorder.shirt_size==='L'
+});
+
+// tried to construct an if function like the MDN example?
+// {
+//   if (backorder.shirt_size === 'L') {
+//   return `Name: ${backorder.first_name} ${backorder.last_name}, email: ${backorder.email}`
+// }});
+// console.log('Challenge 3 Answer: ' + largeShirts);
+console.log('Challenge 3 Answer: ' + JSON.stringify(largeShirts));
 
 // ==== Challenge 4: Use .reduce() ====
 // The donations need to be tallied up and reported for tax purposes. Add up all the donations into a ticketPriceTotal array and log the result
-let ticketPriceTotal = [];
-console.log(ticketPriceTotal);
+let ticketPriceTotal = runners.reduce(function(ticketPriceTotal,currentValue) {return ticketPriceTotal + currentValue.donation},0);
+console.log('Challenge 4 Answer: ' + ticketPriceTotal);
 
 // ==== Challenge 5: Be Creative ====
 // Now that you have used .forEach(), .map(), .filter(), and .reduce().  I want you to think of potential problems you could solve given the data set and the 5k fun run theme.  Try to create and then solve 3 unique problems using one or many of the array methods listed above.
 
 // Problem 1
```

Comment:
good idea. I use to work at a non-profit and being asked to figure something like this out is a great real world problem that you were able to solve with just two lines of code. Well done